src/libstd/sys/unix/os.rs

```diff
@@ -22,15 +22,15 @@ use io;
 use iter;
 use libc::{self, c_int, c_char, c_void};
 use mem;
-use ptr;
 use path::{self, PathBuf};
+use ptr;
 use slice;
 use str;
 use sys::c;
 use sys::fd;
 use vec;
 
-const BUF_BYTES: usize = 2048;
+pub const BUF_BYTES: usize = 2048;
 const TMPBUF_SZ: usize = 128;
 
 fn bytes2path(b: &[u8]) -> PathBuf {
```

Comment:
Similar to what I said above, this and the below `os2path` functions can actually go away now (they're for a time that's long since passed). Could you go ahead and delete them? `bytes2path` can be replaced with `PathBuf::from(OsStr::from_bytes(b))` and `os2path` can be replaced with `PathBuf::from` (e.g. 0-copy)